src/components/FontUploader.tsx

```diff
@@ -1,53 +1,50 @@
-import React, { useState } from 'react';
+import React, { useCallback, useState, useMemo } from 'react';
 import '../styles/FontUploader.css';
+import { useDropzone } from 'react-dropzone';
 
 const FontUploader: React.FC<{ onFontSelected: (selectedFont: File | null) => void }> = ({
   onFontSelected,
 }) => {
   const [, setSelectedFont] = useState<File | null>(null);
   const [, setFontPreview] = useState<string | null>(null);
 
-  const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    if (e.target.files && e.target.files.length > 0) {
-      const file = e.target.files[0];
-      setSelectedFont(file);
-      onFontSelected(file);
-      setFontPreview(URL.createObjectURL(file));
-    }
-  };
-
-  const handleDrop = (e: React.DragEvent<HTMLDivElement>) => {
-    e.preventDefault();
-    const files = Array.from(e.dataTransfer.files);
-
+  const onDrop = useCallback((acceptedFiles: Array<File>) => {
     const allowedExtensions = ['.otf', '.ttf', '.woff', '.woff2'];
-    const fontFiles = files.filter((file) => {
+    const fontFiles = acceptedFiles.filter((file) => {
       const extension = file.name.split('.').pop()?.toLowerCase();
       return allowedExtensions.includes(`.${extension}`);
     });
-
     if (fontFiles.length > 0) {
       const selectedFile = fontFiles[0];
       setSelectedFont(selectedFile);
       onFontSelected(selectedFile);
       setFontPreview(URL.createObjectURL(selectedFile));
     }
-  };
+  }, []);
+
+  const { getRootProps, getInputProps, isDragActive, isFocused } = useDropzone({
+    onDrop,
+    maxFiles: 1,
+  });
 
-  const handleDragOver = (e: React.DragEvent<HTMLDivElement>) => {
-    e.preventDefault();
+  const focusedStyle = {
+    borderColor: '#2196f3',
   };
 
+  const style = useMemo(
+    () => ({
+      ...(isFocused ? focusedStyle : {}),
+    }),
+    [isFocused]
```

Comment:
There was another warning here. Could you just check that out thanks.
![image](https://github.com/typefaceoff/typefaceoff/assets/100255176/fa8690ee-0740-47d9-9413-645e2456e02b)